src/logic/calculate.js

```diff
@@ -0,0 +1,94 @@
+import Big from 'big.js';
+import operate from './operate';
+
+const calculate = (dataObj, buttonName) => {
```

Comment:
**[OPTIONAL]**
const calculate = ({ total, next, operation } , buttonName) => { 
Is equally cleaner

Reply:
This is much cleaner, I didn't realise this is possible. Thanks.